### PR TITLE
PCP: PCPMetric.[ch] Mdoule

### PR DESCRIPTION
**Pull request by smalinux**
Split the PCP Metric API (functions `Metric_*`) into their own module.
as @BenBE suggested.

### CODE REVIEW

**Comment by BenBE** [comment]
Wondering if we should rename those from `Metric_*` to `PCPMetric_*` …

Would better match up with filenames and contained API names.

File: pcp/PCPMetric.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
+/*
+htop - pcp/PCPMetric.c
+(C) 2020-2021 htop dev team
+(C) 2020-2021 Red Hat, Inc.  All Rights Reserved.
```
Comment (BenBE) [inline pcp/PCPMetric.c:4]: Note the copyright header for the other PCP source files.

File: pcp/PCPMetric.h [hunk 2]
```diff
@@ -0,0 +1,179 @@
+#ifndef HEADER_PCPMetric
+#define HEADER_PCPMetric
+/*
+htop - pcp/PCPMetric.h
+(C) 2020-2021 htop dev team
+(C) 2020-2021 Red Hat, Inc.  All Rights Reserved.
```
Comment (BenBE) [inline pcp/PCPMetric.h:6]: Same copyright note …

File: pcp/PCPMetric.h [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+Released under the GNU GPLv2, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include <ctype.h>
+#include <pcp/pmapi.h>
+#include <stdbool.h>
+#include <sys/time.h>
```
Comment (BenBE) [inline pcp/PCPMetric.h:14]: Header order, cf. `STYLEGUIDE`

```suggestion
#include <ctype.h>
#include <stdbool.h>
#include <pcp/pmapi.h>
#include <sys/time.h>
```

File: pcp/PCPProcessList.c [hunk 4]
```diff
@@ -25,6 +26,7 @@ in the source distribution for its full text.
 #include "XUtils.h"
 
 #include "pcp/PCPProcess.h"
+#include "pcp/PCPMetric.h"
```
Comment (BenBE) [inline pcp/PCPProcessList.c:29]: Header order …
```suggestion
#include "pcp/PCPMetric.h"
#include "pcp/PCPProcess.h"
```

**Review by BenBE — CHANGES_REQUESTED** [review]

**Review by natoscott — APPROVED** [review]
LGTM, thanks @smalinux

**Review by BenBE — APPROVED** [review]
LGTM.

**Pull request merged**